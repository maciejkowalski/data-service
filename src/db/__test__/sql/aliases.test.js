const sql = require('../../sql/aliases');

describe('Alias sql query build', () => {
  it('`one` should build select by alias', () => {
    expect(sql.one('scam')).toMatchSnapshot();
  });

  it('`many` should build select by address with order by time and hide blocked by default', () => {
    expect(
      sql.many({ address: '3PELiu6JuVMLJojenRkTPTzCRjAocrxXWJy' })
    ).toMatchSnapshot();
  });

  it('`many` should build select by address with order by time and show blocked if flag is set', () => {
    expect(
      sql.many({
        address: '3PELiu6JuVMLJojenRkTPTzCRjAocrxXWJy',
        showBroken: true,
      })
    ).toMatchSnapshot();
  });
});
