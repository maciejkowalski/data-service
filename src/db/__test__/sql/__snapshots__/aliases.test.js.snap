// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alias sql query build \`many\` should build select by address with order by time and hide blocked by default 1`] = `"select \\"alias\\", \\"address\\", \\"duplicates\\" from (select \\"alias\\" as \\"alias\\", min(\\"sender\\") as \\"address\\", min(\\"time_stamp\\") as \\"time_stamp\\", count(\\"sender\\") as \\"duplicates\\" from \\"txs_10\\" where \\"alias\\" in (select \\"alias\\" from \\"txs_10\\" where \\"sender\\" = '3PELiu6JuVMLJojenRkTPTzCRjAocrxXWJy') group by \\"alias\\") as \\"counted_aliases\\" where \\"duplicates\\" = 1 order by \\"time_stamp\\" asc limit 1000"`;

exports[`Alias sql query build \`many\` should build select by address with order by time and show blocked if flag is set 1`] = `"select \\"alias\\", \\"address\\", \\"duplicates\\" from (select \\"alias\\" as \\"alias\\", min(\\"sender\\") as \\"address\\", min(\\"time_stamp\\") as \\"time_stamp\\", count(\\"sender\\") as \\"duplicates\\" from \\"txs_10\\" where \\"alias\\" in (select \\"alias\\" from \\"txs_10\\" where \\"sender\\" = '3PELiu6JuVMLJojenRkTPTzCRjAocrxXWJy') group by \\"alias\\") as \\"counted_aliases\\" order by \\"time_stamp\\" asc limit 1000"`;

exports[`Alias sql query build \`one\` should build select by alias 1`] = `"select \\"alias\\", \\"address\\", \\"duplicates\\" from (select \\"alias\\" as \\"alias\\", min(\\"sender\\") as \\"address\\", min(\\"time_stamp\\") as \\"time_stamp\\", count(\\"sender\\") as \\"duplicates\\" from \\"txs_10\\" where \\"alias\\" in ('scam') group by \\"alias\\") as \\"counted_aliases\\""`;
